mav_planning_msgs/msg/PolynomialSegment.msg

```diff
@@ -0,0 +1,10 @@
+Header header
+int32 num_coeffs        # order of the polynomial + 1, should match size of x[]
+duration segment_time   # duration of the segment
+float64[] x             # coefficients for the x-axis, INCREASING order
+float64[] y             # coefficients for the y-axis, INCREASING order
+float64[] z             # coefficients for the z-axis, INCREASING order
+float64[] yaw           # coefficients for the yaw, INCREASING order
```

Comment:
I think this message does not need yaw.

Reply:
The idea was to add compatibility for 4dof messages, so it can replace the 4D one, and generalize. We could also just use rz for yaw when rx and ry are empty, would that be better?